### PR TITLE
fixing issue #3811 related with inconsistency with tuist init command

### PR DESCRIPTION
**Pull request by jrescabias**
Resolves https://github.com/tuist/tuist/issues/3811

### Short description 📝

After debugging, I did found that in `InitCommand.swift` at the `preprocess(_ arguments:` function is making a `guard` evaluating if `arguments` contians `"--template"` but not evaluating `"-t"` aswell. I added this and all the cases described above works like a charm.

### How to test the changes locally 🧐

check this init command's variants:
`tuist init`
`tuist init --template default`
`tuist init -t default`
`tuist init --template swiftui` (`Error: Missing expected argument '--name <name>'`)
`tuist init -t swiftui`

### Checklist ✅

- [ ] The code architecture and patterns are consistent with the rest of the codebase.
- [ ] The changes have been tested following the [documented guidelines](https://docs.tuist.io/contributors/testing-strategy/).
- [ ] The `CHANGELOG.md` has been updated to reflect the changes. In case of a breaking change, it's been flagged as such.
- [ ] In case the PR introduces changes that affect users, the documentation has been updated.
- [ ] In case the PR introduces changes that affect how the cache artifact is generated starting from the `TuistGraph.Target`, the `Constants.cacheVersion` has been updated.

### CODE REVIEW

File: Sources/TuistKit/Commands/InitCommand.swift [hunk 1]
```diff
@@ -95,7 +95,8 @@ extension InitCommand {
     static func preprocess(_ arguments: [String]? = nil) throws {
         guard
             let arguments = arguments,
-            arguments.contains("--template")
+            arguments.contains("--template"),
+            arguments.contains("-t")
```
Comment (ezraberch) [inline Sources/TuistKit/Commands/InitCommand.swift:99]: What this guard statement does is check whether a custom template is specified. If not, there's no need to continue. What you've done here is make it so it only continues when the `-t` and `--template` flags are both present. This isn't what we want; the check should continue when either flag is present.

The problem with fixing this bug is that it will break the ability to use a custom template without explicitly specifying the `name` via a command-line argument.

Comment (danyf90) [reply Sources/TuistKit/Commands/InitCommand.swift:99]: Shouldn't it be an or between `arguments.contains("--template")` and `arguments.contains("-t")`? 🤔

Comment (jrescabias) [reply Sources/TuistKit/Commands/InitCommand.swift:99]: > Shouldn't it be an or between `arguments.contains("--template")` and `arguments.contains("-t")`? 🤔

could you rephrase this? I'm not 100 % following. :)

Comment (danyf90) [reply Sources/TuistKit/Commands/InitCommand.swift:99]: I mean there is a template specified if there is a --t OR a --template parameter, while right now the guard it's checking that both are present, isn't it?

Or maybe I'm just missing some context 😁

Comment (jrescabias) [reply Sources/TuistKit/Commands/InitCommand.swift:99]: definitely was my bad 🥲 let see if find the solution then 😄

Comment (fortmarek) [reply Sources/TuistKit/Commands/InitCommand.swift:99]: You should be able to just do `arguments.contains("--template") || arguments.contains("-t")`. Thanks for fixing this btw!

Comment (ezraberch) [reply Sources/TuistKit/Commands/InitCommand.swift:99]: That fixes the bug, but it also means commands of the form `tuist init -t templateName` won't work anymore.

Comment (jrescabias) [reply Sources/TuistKit/Commands/InitCommand.swift:99]: Agree with  @ezraberch , `Missing expected argument '--name <name>'` is shown when using `tuist init -t templateName` or `tuist init --template templateName`

Comment (pepicrft) [reply Sources/TuistKit/Commands/InitCommand.swift:99]: > but it also means commands of the form tuist init -t templateName won't work anymore.

Would you mind expanding on this one? I think I'm lacking some context here.

Comment (ezraberch) [reply Sources/TuistKit/Commands/InitCommand.swift:99]: The way that the `name` attribute is defined in templates is the following:

```swift
let nameAttribute: Template.Attribute = .required("name")
```

Normally, defining something as `.required` means that you must specify a value on the command line. However, there is code which overrides this requirement specifically for the `name` attribute, automatically setting it.

But there is also a second validation check (in [preprocess](https://github.com/tuist/tuist/blob/main/Sources/TuistKit/Commands/InitCommand.swift#L95)), which does not have an exception for the `name` argument. However, this check currently only runs if the `--template` argument exists. So currently `init` works and `init -t foo` works, but `init --template foo` does not.

Fixing this bug in the proposed manner will mean that the check will always run unless you're using the default template, which means you always need to specify `name` explicitly.
